### PR TITLE
feat(oomsotre/stream-cache): make stream push processor configurable

### PR DESCRIPTION
**Pull request by lianxmfor**
resolve: #980 

example:

```yaml
online-store:
  postgres:
    host: 127.0.0.1
    port: 25432
    user: postgres
    password: postgres
    database: oomstore

offline-store:
  postgres:
    host: 127.0.0.1
    port: 25432
    user: postgres
    password: postgres
    database: oomstore

metadata-store:
  postgres:
    host: 127.0.0.1
    port: 25432
    user: postgres
    password: postgres
    database: oomstore

stream-push-processor:
 revision-interval: 24h
 period: 5m
 min-period: 2m
 buffer-interval: 1000
```

### CODE REVIEW

File: pkg/oomstore/cache.go [hunk 1]
```diff
@@ -14,12 +14,12 @@ import (
 	"github.com/oom-ai/oomstore/pkg/oomstore/types"
 )
 
-const (
-	Day       = 24 * time.Hour
-	Capacity  = 1000
-	Period    = 5 * time.Minute
-	MinPeriod = 2 * time.Minute
-)
+var defaultCfg = types.StreamPushProcessorConfig{
```
Comment (wfxr) [inline pkg/oomstore/cache.go:17]: name isolation?

File: pkg/oomstore/types/config.go [hunk 2]
```diff
@@ -26,6 +26,15 @@ type OomStoreConfig struct {
 	MetadataStore MetadataStoreConfig `yaml:"metadata-store"`
 	OfflineStore  OfflineStoreConfig  `yaml:"offline-store"`
 	OnlineStore   OnlineStoreConfig   `yaml:"online-store"`
+
+	StreamPushProcessor *StreamPushProcessorConfig `yaml:"stream-push-processor"`
+}
+
+type StreamPushProcessorConfig struct {
+	RevisionInterval time.Duration `yaml:"revision-interval"`
```
Comment (wfxr) [inline pkg/oomstore/types/config.go:34]: `RevisionInterval` -> `SnapshotInterval` ?

File: pkg/oomstore/types/config.go [hunk 3]
```diff
@@ -26,6 +26,15 @@ type OomStoreConfig struct {
...
+	Period           time.Duration `yaml:"period"`
+	MinPeriod        time.Duration `yaml:"min-period"`
+	Capacity         int           `yaml:"capacity"`
```
Comment (wfxr) [inline pkg/oomstore/types/config.go:37]: `Capacity` -> `BufferSize`

File: pkg/oomstore/types/config.go (hunk 2 above)
Comment (lianxmfor) [reply pkg/oomstore/types/config.go:34]: The `pushStreamProcessor` only controls the revision, the `oomcli snapshot` checks the revision to decide whether to create a new snapshot, so it seems more appropriate to call it revision here 🤔

File: pkg/oomstore/cache.go [hunk 4]
```diff
@@ -14,12 +14,12 @@ import (
...
+	RevisionInterval: 24 * time.Hour,
+	Period:           5 * time.Minute,
+	MinPeriod:        2 * time.Minute,
```
Comment (wfxr) [inline pkg/oomstore/cache.go:20]: I don't think we need both `Period` and `MinPeriod`. Just one `FlushInterval` is enough?

Comment (jinghancc) [reply pkg/oomstore/cache.go:20]: if I remember correctly, `MinPeriod` is to avoid too frequent push

File: pkg/oomstore/cache.go (hunk 1 above)
Comment (lianxmfor) [reply pkg/oomstore/cache.go]: updated

File: pkg/oomstore/types/config.go (hunk 3 above)
Comment (lianxmfor) [reply pkg/oomstore/types/config.go]: updated

File: pkg/oomstore/cache.go (hunk 4 above)
Comment (lianxmfor) [reply pkg/oomstore/cache.go:20]: I think this is open to discussion, but it is not appropriate to change in this PR.

Comment (wfxr) [reply pkg/oomstore/cache.go:20]: @lianxmfor Ok, I can do this refactor later.

File: pkg/oomstore/types/config.go (hunk 3 above)
Comment (wfxr) [reply pkg/oomstore/types/config.go]: typo: BuffSize -> BufferSize

Comment (lianxmfor) [reply pkg/oomstore/types/config.go]: fixed

File: pkg/oomstore/types/config.go (hunk 2 above)
Comment (jinghancc) [reply pkg/oomstore/types/config.go:34]: `RevisionInterval` looks like it applies to batch features as well 🤣

Comment (lianxmfor) [reply pkg/oomstore/types/config.go:34]: `StreamRevisionInterval`?

Comment (wfxr) [reply pkg/oomstore/types/config.go:34]: @lianxmfor This interval is actually for the snapshot service, IMO `SnapshotInterval` is a appropiate name.

Comment (lianxmfor) [reply pkg/oomstore/types/config.go:34]: @wfxr make sense

Comment (wfxr) [reply pkg/oomstore/types/config.go:34]: you can understand it this way: We use `SnapshotInterval` to determin when to create a new revision.

**Comment by lianxmfor** [comment]
I will change the schema of the group in the next PR: add the `snapshpt_interval` field

**Pull request merged**